app/controllers/users/invitations_controller.rb

```diff
@@ -0,0 +1,113 @@
+class Users::InvitationsController < Devise::InvitationsController
+  prepend_before_action :authenticate_inviter!, only: %i[new create]
+  prepend_before_action :has_invitations_left?, only: [:create]
+  prepend_before_action :require_no_authentication, only: %i[edit update destroy]
+  prepend_before_action :resource_from_invitation_token, only: %i[edit destroy]
+
+  helper_method :after_sign_in_path_for if respond_to? :helper_method
+
+  # GET /resource/invitation/new
+  def new
+    self.resource = resource_class.new
+    render :new
+  end
+
+  # POST /resource/invitation
+  def create
+    self.resource = invite_resource
+    resource_invited = resource.errors.empty?
+    yield resource if block_given?
+
+    if resource_invited
+      if is_flashing_format? && resource.invitation_sent_at
+        set_flash_message :notice, :send_instructions, email: resource.email
+      end
+      if method(:after_invite_path_for).arity == 1
+        respond_with resource, location: after_invite_path_for(current_inviter)
+      else
+        respond_with resource, location: after_invite_path_for(current_inviter, resource)
+      end
+    else
+      respond_with_navigational(resource) { render :new }
+    end
+  end
+
+  # GET /resource/invitation/accept?invitation_token=abcdef
+  def edit
+    set_minimum_password_length
+    resource.invitation_token = params[:invitation_token]
+    render :edit
+  end
+
+  # PUT /resource/invitation
+  def update
+    raw_invitation_token = update_resource_params[:invitation_token]
+    self.resource = accept_resource
+    invitation_accepted = resource.errors.empty?
+
+    yield resource if block_given?
+
+    if invitation_accepted
+      if resource.class.allow_insecure_sign_in_after_accept
+        flash_message = resource.active_for_authentication? ? :updated : :updated_not_active
```

Comment:
remove all this, override only those u need to, in other case - call `super` or simply inherit the implementation class
never override the implementation of the gem unless it's necessary!